src/api/music/albums.rs

```diff
@@ -0,0 +1,187 @@
+        }
+
+        if let Some(mincount_val) = mincount {
+            sql += " AND (SELECT COUNT(*) FROM album_tracks WHERE album_tracks.album_id = albums.id) >= ?";
+            params_vec.push(mincount_val.to_string());
+        }
+
+        if let Some(genres_val) = genres {
+            let genres_val = genres_val.into_inner();
+            let genre_placeholders = genres_val.iter().map(|_| "?").collect::<Vec<_>>().join(", ");
+            sql += &format!(" AND album_genres.genre_id IN ({})", genre_placeholders);
+            params_vec.extend(genres_val);
+        }
+
+        sql += &format!(" GROUP BY albums.id LIMIT {}", limit.unwrap_or(50));
+
+        let params_sql: Vec<&dyn ToSql> =
+            params_vec.iter().map(|param| param as &dyn ToSql).collect();
+
+        Ok(Json(
+            conn.prepare(&sql)?
+                .query_map(&params_sql[..], |row| {
+
+                    let artists_str: String = row.get(3)?;
+                    let artists: Vec<String> = artists_str.split(',')
+                        .filter_map(|s| if s.trim().is_empty() { None } else { Some(s.to_string()) })
+                        .collect();
+
+                    let tracks_str: String = row.get(4)?;
+                    let tracks: Vec<String> = tracks_str.split(',')
+                        .filter_map(|s| if s.trim().is_empty() { None } else { Some(s.to_string()) })
+                        .collect();
+
+                    let genres_str: String = row.get(5)?;
+                    let genres: Vec<String> = genres_str.split(',')
+                        .filter_map(|s| if s.trim().is_empty() { None } else { Some(s.to_string()) })
+                        .collect();
+
+                    Ok(Album {
+                        id: row.get(0)?,
+                        name: row.get(1)?,
+                        release: row.get(2)?,
+                        artists,
+                        tracks,
+                        genres,
+                    })
+                })?
+                .map(|v| v.map_err(|e| ApiError::from(e)))
+                .collect::<Result<Vec<Album>>>()?,
+        ))
+    })
+    .await
+}
+
+#[delete("/album/<id>")]
+async fn album_delete(db: Database, user: DangerousUser, id: String) -> Result<()> {
+    if !user.has_permissions(&[Permission::AlbumDelete]) {
+        Err(Status::Forbidden)?
+    }
+
+    db.run(move |conn| -> Result<()> {
+        let tx = conn.transaction()?;
+
+        if let Err(QueryReturnedNoRows) =
+            tx.query_row("SELECT 1 FROM albums WHERE id = ?", params![id], |_| Ok(()))
+        {
+            Err(Status::NotFound)?
+        }
+
+        tx.execute("DELETE FROM albums WHERE id = ?", params![id])?;
+        tx.execute("DELETE FROM album_genres WHERE album_id = ?", params![id])?;
+        tx.execute("DELETE FROM artist_albums WHERE album_id = ?", params![id])?;
+        tx.execute("DELETE FROM album_tracks WHERE album_id = ?", params![id])?;
```

Comment:
These also need to be tested. They are fairly straight forward, so I think that can be left until after the track apis are implemented. Though there is currently no plan for an endpoint that could test the genre album junctions... the joy of using exclusively e2e testing...